nir/src/main/scala/scala/scalanative/nir/Types.scala

```diff
@@ -168,6 +168,8 @@ object Type {
     case Array(tpe, _)      => toArrayClass(tpe)
     case ArrayValue(tpe, _) => toArrayClass(tpe)
     case Function(args, _)  => Global.Top(s"scala.Function${args.length}")
+    case _ =>
+      throw new Exception(s"unknown type ${tpe.show}")
```

Comment:
This could be phrased as "unexpected" instead of "unknown". Essentially, typeToName is used to wrap a type to a name (as per SN terminology). If the match fails, then the type given was not expected there.

Therefore, I would phrase it as something like "type to name: unexpected type ${tpe.show}".

Reply:
Implemented as suggested.  Thank you for the rapid turnaround review and suggested improvement.

When writing C code, I almost always put the function name in the error message, expecially
if there is more than one place where "error foo" might occur. 

It seems the practice is Scala is to omit the method name and let people figure things
out from the Exception stack trace.  I find that takes much more time and emotional energy
to walk back a pig-ugly stack trace than to comprehend a face-evident message.